### DISCUSSION TITLE
Searching specific fields on joined models.

### DISCUSSION THREAD

**Discussion by billyk18278** [Power-Components/livewire-powergrid#795] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

Assuming models are defined properly and datasource is defined using ModelName::query()->with('Submodel1')->with('Submodel2')  
and relationSearch() is defined ['submodel1'=>['sm1field1','sm1field2'], 'submodel2'=>['sm2field1','sm2field2']]  

I have noticed that full table search works perfectly well.(i.e. searching for a value of sm2field1 works)  

but  

 if i want an input search in the specific column (e.g.  ->makeInputText('submodel2.sm2field1') crashes (SQLSTATE[42S22]: Column not found: 1054 Unknown column in 'where clause' )  
I know that if datasource is defined differently [https://github.com/Power-Components/powergrid-demo/blob/main/app/Http/Livewire/DishesTable.php] this is solved but there are obvious advantages in the method of  my example.  

(thanks for the great library anyway)

**Comment by luanfreitasdev** [comment]
Hello!, can you create a test repository that has this situation so I can think of a solution for these cases?. In any case, I can include it in version 4

**Reply by billyk18278** [reply]
thanks for your reply, unfortunately i must deliver a solution soon, but i am sure you understood the "issue", keep it in mind for the next release if you like. To my understanding field specific search using makeInputXX use the core model (relationsearch() is ignored, while tablesearch leverages relationsearch.

**Reply by luanfreitasdev** [reply]
I understand the problem, but I can't simulate it at the moment.

**Comment by billyk18278** [comment]
if you are interested i will try to make i small table in a public url to show you. email me for the temporary link

**Reply by luanfreitasdev** [reply]
If you have the migration, the models and the table component, it's already good

**Reply by billyk18278** [reply]
http://tempgrid.duckdns.org/temp

**Comment by billyk18278** [comment]
`  public function relationSearch(): array
    {
        return [   

            'donor' => [
                'hospital',                     
                'concent',              
                'surname',
                'dob',

             
            ],
            
            
        ];
    }

    /*
    |--------------------------------------------------------------------------
    |  Add Column
    |--------------------------------------------------------------------------
    | Make Datasource fields available to be used as columns.
    | You can pass a closure to transform/modify the data.
    |
    | ❗ IMPORTANT: When using closures, you must escape any value coming from
    |    the database using the `e()` Laravel Helper function.
    |
    */
    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
        // addcolumn custom columns     
            ->addColumn('entrydt')
            ->addColumn('entryclerk')  
            ->addColumn('code')           
            ->addColumn('donor.hospital')            
            ->addColumn('donor.concent')            
            ->addColumn('donor.surname')            
            ->addColumn('donor.dob')
                           
        ;

    }

    /*
    |--------------------------------------------------------------------------
    |  Include Columns
    |--------------------------------------------------------------------------
    | Include the columns added columns, making them visible on the Table.
    | Each column can be configured with properties, filters, actions...
    |
    */

    /**
     * PowerGrid Columns.
     *
     * @return array<int, Column>
     */
    public function columns(): array
    {


        return [
            Column::make(__('CoreModel ENTRY'), 'entrydt')
            ->searchable()
            ->makeInputDatePicker('entrydt')
           
            ->sortable(),
               
            Column::make(__('CoreModel entryclerk'), 'entryclerk')
            ->searchable()
            ->makeInputText('entryclerk'),
            Column::make(__('CoreModel code'), 'code')
            ->searchable()
            ->makeInputText('code'),

            Column::add()
            ->title(__('SubModel concent'))
            ->field('donor.concent','donor.concent')
            ->makeInputSelect(\App\Models\Donor::all(), 'concent', 'concent'),

            Column::make(__('label.donor.surname'), 'donor.surname')
            ->makeInputText('donor.surname'),

            Column::make(__('label.donor.dob'), 'donor.dob')
            ->searchable()
            ->makeInputDatePicker('donor.dob')           
            ->sortable(),
        ];
    }
`

**Reply by billyk18278** [reply]
i am pointing out again, that if i join the tables via join() or DB::RAW  than it works... is when i use ->with() that it does not work.

**Reply by luanfreitasdev** [reply]
https://v4.livewire-powergrid.com/table/column-filters.html#customize-query-collection-return
